packages/contracts-periphery/contracts/universal/DrippieRedd.sol

```diff
@@ -0,0 +1,152 @@
+     */
+    struct DripState {
+        DripStatus status;
+        DripConfig config;
+        uint256 last;
+    }
+
+    /**
+     * Maps from drip names to drip configurations.
+     */
+    mapping(string => DripState) public drips;
+
+    /**
+     * @param _owner Initial owner address.
+     */
+    constructor(address _owner) RetroReceiver(_owner) {}
+
+    /**
+     * Creates a new drip with the given name and configuration.
+     *
+     * @param _name Name of the drip.
+     * @param _config Configuration for the drip.
+     */
+    function create(string memory _name, DripConfig memory _config) public onlyOwner {
+        require(
+            drips[_name].status == DripStatus.NONE,
+            "DrippieRedd: drip with that name already exists"
+        );
+
+        drips[_name] = DripState({ status: DripStatus.ACTIVE, config: _config, last: 0 });
+    }
+
+    /**
+     * Configures a drip by name.
+     *
+     * @param _name Name of the drip to configure.
+     * @param _config Drip configuration.
+     */
+    function update(string memory _name, DripConfig memory _config) public onlyOwner {
+        require(
+            drips[_name].status != DripStatus.NONE,
+            "DrippieRedd: drip with that name does not exist"
+        );
+
+        drips[_name].config = _config;
+    }
+
+    /**
+     * Toggles the status of a given drip.
+     *
+     * @param _name Name of the drip to toggle.
+     */
+    function toggle(string memory _name) public onlyOwner {
+        require(
+            drips[_name].status != DripStatus.NONE,
+            "DrippieRedd: drip with that name does not exist"
+        );
+
+        if (drips[_name].status == DripStatus.ACTIVE) {
+            drips[_name].status = DripStatus.PAUSED;
+        } else {
+            drips[_name].status = DripStatus.ACTIVE;
+        }
+    }
+
+    /**
+     * Triggers a drip.
+     *
+     * @param _name Name of the drip to trigger.
+     */
+    function drip(string memory _name) public {
+        DripState storage state = drips[_name];
+
+        require(
+            state.status == DripStatus.ACTIVE,
+            "DrippieRedd: selected drip does not exist or is not currently active"
+        );
+
+        require(
+            state.last + state.config.interval <= block.timestamp,
+            "DrippieRedd: drip interval has not elapsed"
+        );
+
+        bytes memory checkscript = abi.encodePacked(
+            // Push recipient address onto the stack.
+            hex"73",
+            state.config.recipient,
+            state.config.checkscript,
+            // Checkscript must leave a value on the stack, this cleanup segment will store that
+            // value into memory at positon 0..32 and then revert which allows us to access that
+            // value via returndatacopy below.
+            hex"60005260206000FD"
+        );
+
+        bool ok;
+        assembly {
+            let checker := create(0, add(checkscript, 32), mload(checkscript))
```

Comment:
We gotta make sure that this is worth the costs

Reply:
I can optimize it, just a sec

Reply:
It's about 32000 gas. Might be worth it if we're only transferring once every few days.

Reply:
Ok I went ahead an optimized this by adding a special short-circuit script that avoids the `CREATE` instruction when you specifically just want to do a simple balance threshold check. You can still use checkscripts for more complex stuff (which will be necessary to do the script that funds the Gelato account) but at least that's much less common and will have to run less frequently. Optimized codepath now spends about 10k gas *less* (85k => 75k) than what we currently spend with the multisig transactions that fund these accounts today.